env_vars.tf

```diff
@@ -0,0 +1,45 @@
+variable "service_env_vars" {
+  type        = map(string)
+  default     = {}
+  description = <<EOF
+The environment variables to inject into the service.
+These are typically used to configure a service per environment.
+It is dangerous to put sensitive information in this variable because they are not protected and could be unintentionally exposed.
+EOF
+}
+
+variable "service_secrets" {
+  type        = map(string)
+  default     = {}
+  sensitive   = true
+  description = <<EOF
+The sensitive environment variables to inject into the service.
+These are typically used to configure a service per environment.
+EOF
+}
+
+locals {
+  standard_env_vars = tomap({
+    NULLSTONE_STACK         = data.ns_workspace.this.stack_name
+    NULLSTONE_APP           = data.ns_workspace.this.block_name
+    NULLSTONE_ENV           = data.ns_workspace.this.env_name
+    NULLSTONE_VERSION       = data.ns_app_env.this.version
+    NULLSTONE_COMMIT_SHA    = data.ns_app_env.this.commit_sha
+    NULLSTONE_PUBLIC_HOSTS  = join(",", local.public_hosts)
+    NULLSTONE_PRIVATE_HOSTS = join(",", local.private_hosts)
+  })
+
+  input_env_vars = merge(local.standard_env_vars, local.cap_env_vars, local.app_secret_ids, var.service_env_vars)

```

Comment:
`app_secrets_ids` aren't interpolated, but I'm not sure if that's critical right now.
We don't show these in the UI today.

Reply:
I changed this to not send in the app_secret_ids to the ns_env_variables data source anymore. Instead I am merging those ids into the `all_env_vars` after interpolation is finished.